### PR TITLE
feat: amm trade add shadow request 1%

### PR DESCRIPTION
**Pull request by chef-ryan**
---

## PR-Codex overview
This PR introduces a new experimental feature for optimized AMM trading, updates hooks for trade fetching, and enhances trade API interactions with shadow querying.

### Detailed summary
- Added `OPTIMIZED_AMM_TRADE` to `EXPERIMENTAL_FEATURES`.
- Introduced a new function `useBestTradeFromApiShadow` for optimized AMM trade fetching.
- Updated `useSwapBestOrder` to utilize `useBestTradeFromApiShadow`.
- Modified `useTradeApiPrefetch` to accept `prefix` and `queryType` parameters.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/web/src/config/constants/endpoints.ts [hunk 1]
```diff
@@ -99,3 +99,7 @@ export const WALLET_API = 'https://wallet-api.pancakeswap.com'
 export const BINANCE_DATA_API = 'https://data-api.binance.vision/api'
 
 export const PREDICTION_PRICE_API = '/api/prediction/price'
+
+export const QUOTING_API_PREFIX_ORIGINAL = 'https://pcsx-order-price-api-test-master-viosr.ondigitalocean.app'
+
+export const QUOTING_API_PREFIX_OPTIMIZED = 'https://pcsx-order-price-api-test-branch-nfu7y.ondigitalocean.app'
```
Comment (chefjackson) [inline apps/web/src/config/constants/endpoints.ts:105]: not necessarily to put them in constants as they're mainly for testing purpose

File: apps/web/src/views/Swap/V3Swap/hooks/useSwapBestTrade.ts [hunk 2]
```diff
@@ -58,7 +58,11 @@ export function useSwapBestOrder({ maxHops }: Options = {}) {
     stableSwap: stableSwapEnable,
     trackPerf: true,
     retry: 1,
-  })
+  }
+  const { fetchStatus, data, isStale, error, refetch } = useBestTradeFromApi(bestTradeOptions)
+  useBestTradeFromApiShadow(bestTradeOptions, 'quote-api-ori')
+  useBestTradeFromApiShadow(bestTradeOptions, 'quote-api-opt')
```
Comment (chefjackson) [inline apps/web/src/views/Swap/V3Swap/hooks/useSwapBestTrade.ts:64]: let's enable for all chains. Previously the api is only enabled on arb and eth

File: apps/web/src/hooks/useBestAMMTrade.ts [hunk 3]
```diff
@@ -801,6 +948,38 @@ type PrefetchParams = {
...
+      const res = await serverRes.json()
+      if (!res.success) {
+        throw new Error(res.message)
+      }
+      return res
+    },
+    staleTime: currencyA?.chainId ? POOLS_FAST_REVALIDATE[currencyA.chainId] : 0,
+    refetchInterval: currencyA?.chainId ? POOLS_FAST_REVALIDATE[currencyA.chainId] : 0,
+  })
+}
```
Comment (chefjackson) [inline apps/web/src/hooks/useBestAMMTrade.ts:982]: we don't need a new hook. Let's reuse `useTradeApiPrefetch`

Comment (chefjackson) [reply apps/web/src/hooks/useBestAMMTrade.ts:982]: cuz it's not used right now

File: apps/web/src/hooks/useBestAMMTrade.ts [hunk 4]
```diff
@@ -667,6 +674,146 @@ export function useBestTradeFromApi({
...
+    },
+    prefix,
+    queryType,
+  )
+
+  useEffect(() => {
+    if (featureFlag && !!(amount && currency)) {
+      poolPreFetch.refetch()
+    }
+  }, [amount?.currency, currency])
```
Comment (chefjackson) [inline apps/web/src/hooks/useBestAMMTrade.ts:730]: we don't need refetch for prefetch

File: apps/web/src/config/constants/endpoints.ts (hunk 1 above)
Comment (chef-ryan) [reply apps/web/src/config/constants/endpoints.ts]: done

File: apps/web/src/views/Swap/V3Swap/hooks/useSwapBestTrade.ts (hunk 2 above)
Comment (chef-ryan) [reply apps/web/src/views/Swap/V3Swap/hooks/useSwapBestTrade.ts:64]: done

File: apps/web/src/hooks/useBestAMMTrade.ts (hunk 3 above)
Comment (chef-ryan) [reply apps/web/src/hooks/useBestAMMTrade.ts]: done

File: apps/web/src/hooks/useBestAMMTrade.ts (hunk 4 above)
Comment (chef-ryan) [reply apps/web/src/hooks/useBestAMMTrade.ts]: done

**Review by chefjackson — APPROVED** [review]

**Pull request merged**